homework-selects/sql_questions.sql

```diff
@@ -1,57 +1,937 @@
+
 -- 5. the first and last name and salary for those employees who earn less than the employee earn whose number is 182.
+SELECT
+    first_name,
+    last_name,
+    salary
+FROM
+    employees
+WHERE
+    salary < (
+        SELECT
+            salary
+        FROM
+            employees
+        WHERE
+            employee_id = 182
+    );
+
 -- 6. the first name of all employees including the first name of their manager.
+SELECT
+    e1.first_name,
+    e2.first_name AS manager_first_name
+FROM
+         employees e1
+    INNER JOIN employees e2 ON e1.manager_id = e2.employee_id;
+
+
 -- 7. the first name of all employees and the first name of their manager including those who does not working under any manager.
+SELECT
+    e1.first_name,
+    e2.first_name AS manager_first_name
+FROM
+    employees e1
+    LEFT JOIN employees e2 ON e1.manager_id = e2.employee_id
+                              AND e1.manager_id IS NOT NULL;
+
 -- 8. the details of employees who manage a department.
+SELECT
+    *
+FROM
+    employees e1
+WHERE
+    e1.employee_id IN (
+        SELECT
+            e2.manager_id
+        FROM
+            employees e2
+    )
+    AND e1.department_id IS NOT NULL;
+
 -- 9. the first name, last name, and department number for those employees who works in the same department as the employee who holds the last name as Taylor.
+SELECT
+    e1.first_name,
+    e1.last_name,
+    e1.department_id
+FROM
+    employees e1
+WHERE
+    e1.department_id IN (
+        SELECT
+            e2.department_id
+        FROM
+            employees e2
+        WHERE
+                e2.last_name = 'Taylor'
+            AND ROWNUM = 1
+    );
+
 --10. the department name and number of employees in each of the department.
+SELECT
+    d.department_name,
+    COUNT(e.department_id) AS employees_count
+FROM
+    employees   e,
+    departments d
+WHERE
+    e.department_id = d.department_id
+GROUP BY
+    d.department_name;
+
 --11. the name of the department, average salary and number of employees working in that department who got commission.
+SELECT
+    d.department_name,
+    COUNT(e.department_id) AS employees_count,
+    AVG(e.salary)          AS average_salary
+FROM
+    employees   e,
+    departments d
+WHERE
+        e.department_id = d.department_id
+    AND e.commission_pct IS NOT NULL
+GROUP BY
+    d.department_name;
+
 --12. job title and average salary of employees.
+SELECT
+    j.job_title,
+    AVG(e.salary) AS average_salary
+FROM
+    jobs      j,
+    employees e
+WHERE
+    e.job_id = j.job_id
+GROUP BY
+    j.job_title;
+
 --13. the country name, city, and number of those departments where at least 2 employees are working.
+SELECT
+    l.city,
+    c.country_name,
+    d.department_id,
+    COUNT(d.department_id)
+FROM
+    departments d,
+    employees   e,
+    locations   l,
+    countries   c
+WHERE
+        e.department_id = d.department_id
+    AND d.location_id = l.location_id
+    AND l.country_id = c.country_id
```

Comment:
instead of including these conditions you must introduce them in the FROM clause as here: `FROM countries
         JOIN locations USING (country_id)
         JOIN departments USING (location_id)`